great_expectations/core/batch.py

```diff
@@ -566,9 +566,15 @@ def __init__(
         self._batch_kwargs = batch_kwargs or BatchKwargs()
 
     @property
-    def data(self):
+    def data(self) -> Union[BatchData, pd.DataFrame, SparkDataFrame]:
```

Comment:
If you want you can define an alias for this union. That is somewhere above do, `DataType = Union[BatchData, pd.DataFrame, SparkDataFrame]`, and then you can replace this with `DataType`.

Reply:
This is more of a nit but isn't the property here superfluous?

It doesn't really protect `self._data` from access or modification, and it's not calculating anything.